docs/develop/typescript/message-passing.mdx

```diff
@@ -122,14 +122,11 @@ export async function greetingWorkflow(): Promise<string> {
 - The handler cannot return a value.
   The response is sent immediately from the server, without waiting for the Workflow to process the Signal.
 
-- Signal (and Update) handlers can be `async`.
+- Signal and Update handlers can be `async`.
   This allows you to use Activities, Child Workflows, durable [`workflow.sleep`](https://typescript.temporal.io/api/namespaces/workflow#sleep) Timers, [`workflow.condition`](https://typescript.temporal.io/api/namespaces/workflow#condition) conditions, and more.
   See [Async handlers](#async-handlers) and [Workflow message passing](/encyclopedia/workflow-message-passing) for guidelines on safely using async Signal and Update handlers.
 
-- Delay calling [`workflow.setHandler`](https://typescript.temporal.io/api/namespaces/workflow#sethandler) until the Workflow initialization needed by Signal (or Update) handlers has finished.
-  This is safe because the SDK buffers messages when there are no registered handlers for them.
-  Note that [`workflow.setHandler`](https://typescript.temporal.io/api/namespaces/workflow#sethandler) will immediately invoke the handler of buffered Signals (or Updates) with matching types.
-  This could lead to out-of-order processing of messages with different types.
+- If your Workflow needs to do some async initialization before handling a Signal or Update, use [`workflow.condition`](https://typescript.temporal.io/api/namespaces/workflow#condition) inside your handler to wait until initialization has completed.
```

Comment:
@mjameswh I've removed the statement about `setHandler` immediately invoking buffered messages because I think it's too technical to explain buffering and diffrerences therein between Signal and Update in the docs. However, would it make sense to say something like

> You should typically call `setHandler` at the start of your Workflow.

?

Reply:
> However, would it make sense to say something like
>> You should typically call setHandler at the start of your Workflow.

I don't think that would be useful.

> I've removed the statement about setHandler immediately invoking buffered messages because I think it's too technical to explain buffering and differences therein between Signal and Update in the docs

I have had to explain that loophole a few times, so I would like to have that information somewhere that users can find, though I totally agree this is a bit too much for this specific section. Let's remove it from here, and I'll find some better place for that.